patches/net/minecraft/network/RegistryFriendlyByteBuf.java.patch

```diff
@@ -0,0 +1,43 @@
+--- a/net/minecraft/network/RegistryFriendlyByteBuf.java
++++ b/net/minecraft/network/RegistryFriendlyByteBuf.java
+@@ -6,16 +_,40 @@
+ 
+ public class RegistryFriendlyByteBuf extends FriendlyByteBuf {
+     private final RegistryAccess registryAccess;
++    private final net.neoforged.neoforge.network.connection.ConnectionType connectionType;
+ 
++    /**
++     * @deprecated Neo: use overload with ConnectionType context
```

Comment:
I wouldn't make it deprecated cause mods might want to use this. Can't we make it default to `MODDED`?

Reply:
It was defaulted to `NEOFORGE` in the initial commits and changed on XFactHD's request.
I was not quite sure about what the default should be at the time, but since this is brought up again, might need XFactHD to elaborate on the change request.
I believe defaulting to `NEOFORGE` and remove deprecation is fine as all vanilla consumers of the constructor are well patched with correct connection type and mods using this will likely to be needed on both sides.

Reply:
Defaulting to `OTHER` is most appropriate here.  When we add additional fields, we deprecate the original and provide a vanilla-compatible default.  Mods can use the other constructor and explicitly provide the type, since they may need to retrieve it from the current context rather than encode it statically.

Reply:
> When we add additional fields, we deprecate the original and provide a vanilla-compatible default.

The motivation of leaving *vanilla-compatible* default are:

- We usually don't patch vanilla consumers of the method to provide a non-default context, the extension is for modders.
- (Multiplatform) mods are safe to compile against and run with the vanilla method if they don't need the extension.

However, this is **not** the case for this patch for RFBB, the extension is more for NeoForge's internal implementation, and vanilla consumers are well patched using the extension instead of a default.

Mods should be free to use either methods, but in most case they would want the default one with `NEOFORGE` because most mods does not supports connecting to vanilla, and thus only for rare cases they are sensitive to connection. We won't want multiplatform mods to use `OTHER` just because they compiled against the vanilla method in common code, these mods are likely not to be sending things to `OTHER` at runtime and we don't want them to lose data for not using `NEOFORGE`.

Do note that mods usually creates `StreamCodec` instead of `RFBB` instance, they are free to provide a connection aware `StreamCodec` through `NeoForgeStreamCodecs#connectionAware` if they really need it.

Reply:
> Mods should be free to use either methods, but in most case they would want the default one with NEOFORGE because most mods does not supports connecting to vanilla, and thus only for rare cases they are sensitive to connection.

That is my line of reasoning as well. :)

Reply:
> We usually don't patch vanilla consumers of the method to provide a non-default context, the extension is for modders.

That is not true, vanilla callsites are patched in almost all cases to call the extension (including this case), otherwise the extensions would be entirely useless.

> Mods should be free to use either methods, but in most case they would want the default one with `NEOFORGE` because most mods does not supports connecting to vanilla, and thus only for rare cases they are sensitive to connection

There are cases where you need a context-sensitive connection type because a downstream `StreamCodec` you call into might be using the type-awareness and there are also mods which support cross-loader connections (which will most likely be constructing an RFBB from a loader-agnostic source set), in which case `OTHER` or a context-sensitive type must be used. For those reasons it is in my opinion preferable to fall back to a known safe option if the type is not explicitly specified and make people aware that they should provide an explicit type by making the original constructor and factory deprecated.

Reply:
The two approaches are reasonable. Being more cautious is not a bad idea.